src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -250,10 +271,11 @@ void PerformBatchedCompletionTask(IMessageReceiver internalReceiver, int index)
                         await HandleCompletion(message, context, completionCanBeBatched).ConfigureAwait(false);
                     }
                 }
-                catch (Exception onErrorException)
-                {
-                    await AbandonAsync(message, onErrorException).ConfigureAwait(false);
-                }
+            }
+            catch (Exception onErrorException)
+            {
+                await AbandonAsync(message, onErrorException).ConfigureAwait(false);
```

Comment:
If the recoverability block fails in `AbandonAsync` we're going to try `AbandonAsync` again.